### PR TITLE
LSR-1069 Update to Swift 4

### PR DESCRIPTION
**Pull request by victorg1991**

### CODE REVIEW

**Comment by victorg1991** [comment]
Ignore Lint issues, we have to make another PR linting the whole library

**Comment by dgarciasarai** [comment]
I made a rebase to fix conflics and solve SwiftLint issues.

File: ios/Framework/Core/DDL/FormScreenlet/Connectors/DDLFormRecordLoadLiferayConnector.swift [hunk 1]
```diff
@@ -80,8 +80,8 @@ open class Liferay62DDLFormRecordLoadConnector: DDLFormRecordLoadLiferayConnecto
 	override open func getRecord(_ session: LRSession, recordId: Int64, locale: String) throws -> [NSObject: AnyObject] {
 		let service = LRScreensddlrecordService_v62(session: session)
 
-		return try service?.getDdlRecord(withDdlRecordId: recordId,
-		locale: NSLocale.currentLocaleString) as! [NSObject: AnyObject]
+		return try service!.getDdlRecord(withDdlRecordId: recordId,
```
Comment (dgarciasarai) [inline ios/Framework/Core/DDL/FormScreenlet/Connectors/DDLFormRecordLoadLiferayConnector.swift]: Why this?

File: ios/Samples/AndorraTelecomPublic/Podfile [hunk 2]
```diff
@@ -11,6 +12,9 @@ end
 post_install do |installer|
 	installer.pods_project.targets.each do |target|
 		target.build_configurations.each do |config|
+			if target.to_s == 'Hokusai'
+				config.build_settings['SWIFT_VERSION'] = '3.2'
```
Comment (dgarciasarai) [inline ios/Samples/AndorraTelecomPublic/Podfile]: It seems that `Hokusai` library provides support to Swift 4, isn't it?

File: ios/Samples/WesterosEmployees-hybrid/Theme/Asset/Display/AssetDisplayView_westeros.swift [hunk 3]
```diff
@@ -1,22 +0,0 @@
...
- *
- * This library is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
- * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
- * details.
- */
-import UIKit
-import LiferayScreens
-
-open class AssetDisplayView_westeros: AssetDisplayView_default {
```
Comment (dgarciasarai) [inline ios/Samples/WesterosEmployees-hybrid/Theme/Asset/Display/AssetDisplayView_westeros.swift]: Why all those theme files were removed?

**Comment by dgarciasarai** [comment]
LGTM 👍 🎉  Please, review the comments before merge

File: ios/Samples/WesterosEmployees-hybrid/Theme/Asset/Display/AssetDisplayView_westeros.swift (hunk 3 above)
Comment (victorg1991) [reply ios/Samples/WesterosEmployees-hybrid/Theme/Asset/Display/AssetDisplayView_westeros.swift]: They are not used in this demo and they are repeated

File: ios/Samples/AndorraTelecomPublic/Podfile (hunk 2 above)
Comment (victorg1991) [reply ios/Samples/AndorraTelecomPublic/Podfile]: Not in the momment I tried that. I think the conversion is badly done

**Pull request merged**